src/natcap/invest/cli.py

```diff
@@ -587,7 +587,13 @@ def main(user_args=None):
 
             # We're deliberately not validating here because the user
             # can just call ``invest validate <datastack>`` to validate.
-            getattr(model_module, 'execute')(parsed_datastack.args)
+            try:
+                getattr(model_module, 'execute')(parsed_datastack.args)
+            # Graceful exit from this program, otherwise the pyinstaller-made
+            # executeable will exit with 'Failed to execute script cli' on an
+            # uncaught exception. And that only adds unhelpful noise to stderr.
+            except Exception as error:
+                parser.exit(DEFAULT_EXIT_CODE, str(error))
```

Comment:
Wouldn't this also capture and silence the exception itself?  Would it be useful to add `LOGGER.exception(...)`?

Reply:
Hmm the exception still ends up in the logfile, but now that you mention it, I'm not sure why.

Reply:
Oh okay it looks like you're exactly right and this explains the difference in the tracebacks, where everything missing from the **after** example  up top is the traceback that's being captured here.